### PR TITLE
WIP: Scan Loop Working - With Notebook

### PR DESCRIPTION
**Pull request by chancehaycock**

### CODE REVIEW

File: scan.py [hunk 1]
```diff
@@ -86,44 +90,95 @@ def EBP(forecast_data: pd.DataFrame, grid_partition: int,
...
+        region_col_scores = []
+        for i in range(len(x_ticks) - 1):
+
+            sub_df = res_df[
+                (res_df["x_min"] <= x_ticks[i])
+                & (res_df["x_max"] >= x_ticks[i + 1])
+                & (res_df["y_min"] <= y_ticks[j])
+                & (res_df["y_max"] >= y_ticks[j + 1])
+            ]
+            region_score = sub_df["likelihood_score"].sum() / len(
```
Comment (TeddyTW) [inline scan.py:162]: okay so this sums over the likelyhood regions for each grid locations and then divides by the number of likelyhood regions that the specific grid space is contained within (sor of mean over likelyhoood regions)

File: timeseries.py [hunk 2]
```diff
@@ -288,7 +288,7 @@ def CB_plot(df: pd.DataFrame):
     df_format["hour_from_start"] = df_format["hour_from_start"].astype(
         dtype="timedelta64[h]"
     )
-    offset = colors.DivergingNorm(vmin=0.5, vcenter=1, vmax=2)
```
Comment (TeddyTW) [inline timeseries.py:291]: should do the same thing? guess we shall find out ahaha

File: region.py [hunk 3]
```diff
@@ -137,18 +142,102 @@ def make_grid(global_region: Type[Region], N: int) -> tuple:
...
+        t_min = res_df["t_min"].min()
+        t_max = res_df["t_max"].max()
+    else:
+        t_min = res_df.iloc[rank].t_min
+        t_max = res_df.iloc[rank].t_max
+    return Region(x_min, x_max, y_min, y_max, t_min, t_max)
+
+
+# Plot the time series of all detectors within a region of interest
+def plot_region_time_series(region: Type[Region], forecast_df: pd.DataFrame) -> None:
```
[the remark this answers is not in the document]
Comment (chancehaycock) [reply region.py:220]: Yeah - makes sense!

File: scan.py [hunk 4]
```diff
@@ -44,18 +48,24 @@ def EBP(forecast_data: pd.DataFrame, grid_partition: int,
...
     ]
     region_score_df = pd.DataFrame(columns=columns)
 
+    num_regions = 0
     # Loop over all possible prisms in the space
     for i, _ in enumerate(x_ticks):
         for j in range(i + 1, len(x_ticks)):
             for k, _ in enumerate(y_ticks):
                 for l in range(k + 1, len(y_ticks)):
                     for t in range(1, len(t_ticks)):
```
[the remark this answers is not in the document]
Comment (chancehaycock) [reply scan.py:64]: Yeah that's right

File: scan.py (hunk 1 above)
Comment (chancehaycock) [reply scan.py:162]: I'll try to formulate this well:
 - We have a N x N grid. I start by looping over all of the N^2 cells.
- For each cell, I check the results dataframe and ask, "Hey, find all the space-time regions that I searched over in the main loop that include my little cell."
- (The middle cell will have the most of these available)
- I then assign the average likelihood score that cell attained to it. (One of the space-time regions will be the cell itself, others are all supersets of the cell.
- Plot the results!

File: timeseries.py (hunk 2 above)
Comment (chancehaycock) [reply timeseries.py:291]: Change it back it breaks yours. I just had a Deprecation Warning

File: scan.py (hunk 1 above)
Comment (chancehaycock) [reply scan.py:162]: I should add - there are proabbly another 500 suitable plots for displaying this info. Let me know if you make one.

Comment (TeddyTW) [reply scan.py:162]: Okayyyy makes sense. and then you divide by the number of regions that the cell is contained within to remove the effects of the middle cell?

**Pull request merged**